installers/common/MacInstaller.hs

```diff
@@ -96,6 +114,39 @@ main opts@Options{oBackend, oCluster, oBuildJob, oOutputDir, oTestInstaller, oSi
         NotSigned -> rm opkg
     Nothing -> pure ()
 
+-- | Define the code signing script to be used for code signing
+codeSignScriptContents :: String
+codeSignScriptContents = [r|#!/run/current-system/sw/bin/bash
+set -x
+SIGN_ID="$1"
+KEYCHAIN="$2"
+REL_PATH="$3"
+ABS_PATH="$(pwd)/$REL_PATH"
+SIGN_CMD="codesign --verbose=4 --deep --strict --timestamp --options=runtime --sign \"$SIGN_ID\""
+VERIFY_CMD="codesign --verbose=4 --verify --deep --strict"
+TS="$(date +%Y-%m-%d_%H-%M-%S)"
+LOG="2>&1 | tee -a /tmp/codesign-output-${TS}.txt"
+
+# Remove symlinks pointing outside of the project build folder:
+rm -f "$ABS_PATH/Contents/Resources/app/result"
+
+# Ensure the code signing identity is found and set the keychain search path:
+eval "security show-keychain-info \"$KEYCHAIN\" $LOG"
+eval "security find-identity -v -p codesigning \"$KEYCHAIN\" $LOG"
+eval "security list-keychains -d user -s \"$KEYCHAIN\" $LOG"
+
+# Sign framework executables not signed by the deep sign command:
+eval "$SIGN_CMD \"$ABS_PATH/Contents/Frameworks/Squirrel.framework/Versions/A/Resources/ShipIt\" $LOG"
+eval "$SIGN_CMD \"$ABS_PATH/Contents/Frameworks/Electron Framework.framework/Versions/Current/Resources/crashpad_handler\" $LOG"
+eval "$SIGN_CMD \"$ABS_PATH/Contents/Frameworks/Electron Framework.framework/Versions/Current/Libraries/libnode.dylib\" $LOG"
+eval "$SIGN_CMD \"$ABS_PATH/Contents/Frameworks/Electron Framework.framework/Versions/Current/Libraries/libffmpeg.dylib\" $LOG"
+
+# Sign the whole component deeply
+eval "$SIGN_CMD \"$ABS_PATH\" $LOG"
+eval "$VERIFY_CMD \"$ABS_PATH\" $LOG"
+eval "$VERIFY_CMD --display -r- \"$ABS_PATH\""
+set +x|]
+
```

Comment:
maybe this should be a shell script that nix just adds to `$PATH` for us?

Reply:
I'm ok leaving here from now and moving it out if we add any additional complexity in the future.